src/java.naming/share/classes/com/sun/jndi/ldap/AbstractLdapNamingEnumeration.java

```diff
@@ -94,10 +127,10 @@
                 refEx = answer.refEx;
             }
 
+            this.state = new CleaningAction(homeCtx, answer, homeCtx.clnt);
             // Ensures that context won't get closed from underneath us
-            this.homeCtx = homeCtx;
-            homeCtx.incEnumCount();
-            enumClnt = homeCtx.clnt; // remember
+            this.state.homeCtx.incEnumCount();
```

Comment:
Readability might be improved by using the new `homeCtx()` method instead of `state.homeCtx` in this file.
It would then be consistent with how subclasses access it.
It depends which style you prefer to be more consistent with.